### PR TITLE
Adding new Voting UI

### PR DESCRIPTION
**Pull request by JosueG15**
Screen Shot 2021-07-08 at 18 06 19
Screen Shot 2021-07-08 at 18 06 26
Screen Shot 2021-07-08 at 18 06 33
Screen Shot 2021-07-08 at 18 05 59
Screen Shot 2021-07-08 at 18 06 06
Screen Shot 2021-07-08 at 18 06 13

## Removed
- Removed Old Ranked Files

## Added 
- New RankedChoice Component
- Added RankedChoice Context.
- Added new schemas for quizzes on voting
- Added clientData Key for AboutDataSharing to get consentModel info

## Modified
- Modified SageForm to add new schemas.
- Modified All ArmFlows to add RankedChoice Component by props

### CODE REVIEW

File: src/components/dashboard/armFlows/ArmFlowFour.tsx [hunk 1]
```diff
@@ -105,13 +113,12 @@ function ArmFlowFour({
               subTitle={t('form.armFour.pageTwo.subTitle')}
               errorMessage={errorMessage}
               formId={FORM_IDS.HOW_RESEARCHERS_ACCESS}
-              onSubmit={(event: any) => {
+              onSubmit={async (event: any) => {
```
Comment (jwilliam706) [inline src/components/dashboard/armFlows/ArmFlowFour.tsx]: There's no need for this to be async

File: src/components/dashboard/armFlows/ArmFlowFour.tsx [hunk 2]
```diff
@@ -7,26 +7,30 @@ import {
...
+  handleNext: (fields?: object) => void
   handleBack: () => void
-  handleComplete: (pageId?: string) => void
+  handleComplete: (fields?: object) => void
   updateClientData: Function
+  RankedChoice: any
+  clientData: any
 }
 
+const maxSteps = 10
```
Comment (jwilliam706) [inline src/components/dashboard/armFlows/ArmFlowFour.tsx]: Are we using this variable? 
If we are, we already have a getArmFlowMaxSteps function, there's no need for duplication.

File: src/components/dashboard/armFlows/ArmFlowOne.tsx [hunk 3]
```diff
@@ -17,29 +17,26 @@ import { ReactComponent as Globe } from 'assets/consent/globe.svg'
...
-  maxSteps: number
-  handleNext: (pageId?: string) => void
+  handleNext: (fields?: object) => void
   handleBack: () => void
-  handleComplete: (pageId?: string) => void
+  handleComplete: (fields?: object) => void
   updateClientData: Function
 }
 
+const maxSteps = 3
```
Comment (jwilliam706) [inline src/components/dashboard/armFlows/ArmFlowOne.tsx]: pass as prop if used

File: src/components/dashboard/armFlows/ArmFlowThree.tsx [hunk 4]
```diff
@@ -6,53 +6,88 @@ import {
...
-  handleNext: (pageId?: string) => void
+  handleNext: (fields?: object) => void
   handleBack: () => void
-  handleComplete: (pageId?: string) => void
+  handleComplete: (fields?: object) => void
   updateClientData: Function
+  RankedChoice: any
 }
 
+const maxSteps = 9
```
Comment (jwilliam706) [inline src/components/dashboard/armFlows/ArmFlowThree.tsx]: pass as a prop

File: src/components/dashboard/armFlows/ArmFlowTwo.tsx [hunk 5]
```diff
@@ -6,53 +6,84 @@ import {
...
-  handleNext: (pageId?: string) => void
+  handleNext: (fields?: object) => void
   handleBack: () => void
-  handleComplete: (pageId?: string) => void
+  handleComplete: (fields?: object) => void
   updateClientData: Function
+  RankedChoice: any
 }
 
+const maxSteps = 9
```
Comment (jwilliam706) [inline src/components/dashboard/armFlows/ArmFlowTwo.tsx]: pass as a prop

File: src/data/hub/hub.ts [hunk 6]
```diff
@@ -22,7 +22,7 @@ export const hubSteps: HubCardProps[] = [
     title: 'About the study',
     subtitle: 'Learn about the study and the risks and benefits of joining',
     time: 'Approx. 7 minutes',
-    status: 'disabled',
```
Comment (jwilliam706) [inline src/data/hub/hub.ts]: This should remain disabled.

File: src/components/dashboard/armFlows/ArmFlowFour.tsx (hunk 1 above)
Comment (JosueG15) [reply src/components/dashboard/armFlows/ArmFlowFour.tsx]: @jwilliam706 Done.

**Review by jwilliam706 — CHANGES_REQUESTED** [review]
Please review the comments

File: src/components/dashboard/armFlows/ArmFlowOne.tsx (hunk 3 above)
Comment (JosueG15) [reply src/components/dashboard/armFlows/ArmFlowOne.tsx]: @jwilliam706 done

File: src/components/dashboard/armFlows/ArmFlowThree.tsx (hunk 4 above)
Comment (JosueG15) [reply src/components/dashboard/armFlows/ArmFlowThree.tsx]: @jwilliam706 done

File: src/components/dashboard/armFlows/ArmFlowTwo.tsx (hunk 5 above)
Comment (JosueG15) [reply src/components/dashboard/armFlows/ArmFlowTwo.tsx]: @jwilliam706 done

File: src/data/hub/hub.ts (hunk 6 above)
Comment (JosueG15) [reply src/data/hub/hub.ts]: @jwilliam706 updated.

File: src/components/dashboard/armFlows/ArmFlowFour.tsx (hunk 2 above)
Comment (JosueG15) [reply src/components/dashboard/armFlows/ArmFlowFour.tsx]: @jwilliam706 done.

**Pull request merged**